### PR TITLE
Added exponential Search

### LINKED ISSUE #187: Exponential search
Would like to state this with example Let me know

### PR DESCRIPTION
**Pull request by gopichandpuli9**
## Related Issuse
  - Info about Issue or bug

Closes: #[issue number that will be closed through this PR]

#### Describe the changes you've made
A clear and concise description of what you have done to successfully close your assigned issue. Any new files? or anything you feel to let us know!

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots

 Original           | Updated
 :--------------------: |:--------------------:
 **original screenshot**  | <b>updated screenshot </b> |

### CODE REVIEW

**Comment by ankitapuri** [comment]
your extension is wrong its cpp .
Also add comments explaining your code in detail .

**Comment by gopichandpuli9** [comment]
Done the necessary changes

File: Search_Techniques/exponential_search.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+   }
+   return binarySearch(array, i/2, i, key); //search item in the smaller range
+}
+
+int main() {
+   int n, searchKey, loc;
+   cout << "Enter number of items: "; // number of elements where search has to be made
+   cin >> n;
+   int arr[n]; //create an array of size n
+   cout << "Enter items: " << endl; //input the elements of the array
```
Comment (ankitapuri) [inline Search_Techniques/exponential_search.cpp:35]: mention to enter array in sorted manner.

**Comment by gopichandpuli9** [comment]
Added the statement

**Review by ankitapuri — APPROVED** [review]

**Pull request merged**